src/ruby/lib/grpc/generic/active_call.rb

```diff
@@ -205,6 +205,8 @@ def remote_send(req, marshalled = false)
     # list, mulitple metadata for its key are sent
     def send_status(code = OK, details = '', assert_finished = false,
                     metadata: {})
+      return if response_status_as_trailers_only(code, details, metadata: metadata)
+
```

Comment:
I don't think we need the new `response_status_as_trailers_only` utility. If below, instead of calling `send_initial_metadata`, we first checked whether or not we've already sent initial metadata and then added the send initial metadata op to the same batch as the `send_status_from_server` op, then I think that would do what's needed here in a slightly simpler way.

Reply:
fixed. https://github.com/grpc/grpc/pull/16490/commits/6c1e8d34c3b11b96c0f4c91559940426065e09ef